lib/vector-core/src/event/metric/arbitrary.rs

```diff
@@ -0,0 +1,113 @@
+use proptest::{collection::btree_set, prelude::*};
+
+use crate::metrics::AgentDDSketch;
+
+use super::{Bucket, MetricSketch, MetricValue, Quantile, Sample, StatisticKind};
+
+fn arb_metric_value() -> BoxedStrategy<MetricValue> {
+    let strategy = prop_oneof![
+        any::<f64>().prop_map(|value| MetricValue::Counter { value }),
+        any::<f64>().prop_map(|value| MetricValue::Gauge { value }),
+        btree_set("[a-z0-9]{8,16}", 2..16).prop_map(|values| MetricValue::Set { values }),
+        any::<(Vec<Sample>, StatisticKind)>()
+            .prop_map(|(samples, statistic)| MetricValue::Distribution { samples, statistic }),
+        any::<(Vec<Bucket>, u64, f64)>().prop_map(|(buckets, count, sum)| {
+            MetricValue::AggregatedHistogram {
+                buckets,
+                count,
+                sum,
+            }
+        }),
+        any::<(Vec<Quantile>, u64, f64)>().prop_map(|(quantiles, count, sum)| {
+            MetricValue::AggregatedSummary {
+                quantiles,
+                count,
+                sum,
+            }
+        }),
```

Comment:
Is it valid for these values to really be _any_ here? Ie, `count` and `sum` would normally be related to the contents of the `buckets`/`quantiles`.

Reply:
So, good eye, it's not!

I basically scaffolded out the `Arbitrary` implementations just enough to get the property test running, but I can go back and fix this up quickly, since it's not much work to do so.

Reply:
Sorted this out in 9179db442.